### ISSUE TITLE
Apple Pay button does not conform to Apple's guidelines

### ISSUE THREAD

**Issue by ejensen** [braintree/braintree-ios-drop-in#231]
**Integration Details (please complete the following information):**
 - SDK/Library Version: 8.1.1
 - Environment: All
 - iOS Version: All
 - Device: All
 - Integration type & version: All

**Describe the bug**
The Braintree Drop-in UI displays the [Apple Pay mark ](https://developer.apple.com/design/human-interface-guidelines/apple-pay/overview/buttons-and-marks/#apple-pay-mark) as a payment method button. Apple's [Human Interface Guidelines for Apple Pay](https://developer.apple.com/design/human-interface-guidelines/apple-pay/overview/buttons-and-marks/) explicitly states that as something that should **not** be done:

> Use the Apple Pay mark only to communicate that Apple Pay is accepted. The mark doesn’t facilitate payment. Never use it as a payment button or position it as a button.
From https://developer.apple.com/design/human-interface-guidelines/apple-pay/overview/offering-apple-pay/

Apple's [Human Interface Guidelines for Apple Pay](https://developer.apple.com/design/human-interface-guidelines/apple-pay/overview/buttons-and-marks/) states that an Apple Pay button ([`PKPaymentButton`](https://developer.apple.com/documentation/passkit/pkpaymentbutton)) should be used to initiate Apple Pay payments:

> The Apple Pay mark is not a button; if you need an Apple Pay button, choose one of the buttons described in [Button Types](https://developer.apple.com/design/human-interface-guidelines/apple-pay/overview/buttons-and-marks/#button-types). Use the Apple-provided APIs to create Apple Pay buttons.

The non-conformance to the Human Interface guidelines is causing apps to be flagged by Apple in the review process.

**Expected behavior**
The Braintree Drop-in UI displays Apple Pay in a way that conforms to the [Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/apple-pay/overview/buttons-and-marks/).

One possible solution is to change all the payment options have button appearances. This change would not only resolve the issue with Apple Pay's design guidelines but would also improve conformance with the [design guidelines for offering PayPal and Venmo](https://developer.paypal.com/docs/checkout/integration-features/customize-button/) and [Google Pay](https://developers.google.com/pay/api/android/guides/brand-guidelines)

**Screenshots**
_Current Drop-in UI showing the inappropriate usage of the Apple Pay Mark:_
![current](https://user-images.githubusercontent.com/915431/90554915-65817080-e14b-11ea-89b2-7fc335049948.png)

_Proposed solution that uses [Apple's approved button](https://developer.apple.com/design/human-interface-guidelines/apple-pay/overview/buttons-and-marks/#button-types) and [PayPal's recommended button](https://developer.paypal.com/docs/checkout/integration-features/customize-button/#color):_
![suggested](https://user-images.githubusercontent.com/915431/90555116-ba24eb80-e14b-11ea-83fe-b680ba37fb65.png)

**Comment by scannillo** [comment]
Hi @ejensen - thank you for opening this issue and providing detail/references. We will take a look at this issue as soon as we can, and keep you posted here.

**Comment by adamfellon** [comment]
Hi @ejensen , did you figure out a way to solve this?   I feel like the `BTPaymentButton` makes similar configuration concessions for other platforms, so maybe we can just flip the `backgroundColor` or something?

Asking because I was rejected today by Apple for the button color & background provided in the SDK.

Thanks!

**Comment by scannillo** [comment]
Hi @adamfellon can you provide the information provided to you by Apple regarding this Apple Payment button rejection?

**Comment by adamfellon** [comment]
Yep!

> Guideline 3.1.6 - Business - Payments - Apple Pay


> Your app uses Apple Pay as a purchasing mechanism but does not use Apple Pay branding and user interface elements appropriately as described in the Apple Pay Human Interface Guidelines. Specifically:

> - You use the Apple Pay mark as a button, rather than one of the approved Apple Pay button designs.

> Next Steps

> Please revise Apple Pay implementation in your app so that it is compliant with all the App Store Review Guidelines and the Apple Pay Human Interface Guidelines.

> Resources

> For guidance on how to use Apple Pay button assets and payment marks in the customer purchase flow, review the Apple Pay Human Interface Guidelines.

> Please see attached screenshot for details.

That "apple button designs" is a link out to here: https://developer.apple.com/design/human-interface-guidelines/apple-pay/overview/buttons-and-marks/

Right now we're trying to apply the dark theme, and hoping that their main complaints are based on white button on lightGrey background.  Any help or advice would be appreciated, thanks!

**Comment by scannillo** [comment]
@adamfellon Thanks for giving us those details, it helps a lot. We'll take a look at this ASAP.

Also to confirm, you're using v8.1.1 of Braintree iOS Drop In?

**Comment by adamfellon** [comment]
From our Podfile.lock , looks like we use 8.1.2

```
  - BraintreeDropIn (8.1.2):
    - BraintreeDropIn/DropIn (= 8.1.2)
  - BraintreeDropIn/DropIn (8.1.2):
    - Braintree/Card (~> 4.32)
    - Braintree/Core (~> 4.32)
    - Braintree/PaymentFlow (~> 4.32)
    - Braintree/PayPal (~> 4.32)
    - Braintree/UnionPay (~> 4.32)
    - BraintreeDropIn/UIKit
  - BraintreeDropIn/UIKit (8.1.2)
```

**Comment by sestevens** [comment]
@adamfellon Thanks for confirming! We're working on an update that fixes how the Apple Pay button is displayed. In the meantime, if you need a _temporary_ workaround, you could display the Apple Pay button next to your checkout button (or whatever button triggers the presentation of Drop-in), and then hide the Apple Pay row in Drop-in by setting [`applePayDisabled`](https://github.com/braintree/braintree-ios-drop-in/blob/96a7e365c5a0771e19f6d05bcbf8d076964942d3/BraintreeDropIn/Public/BTDropInRequest.h#L21) to `true` on your `BTDropInRequest`.

Of course, the long-term solution involves an update to this SDK. We'll post a comment in this issue when we release a new version with the fix.

**Comment by adamfellon** [comment]
Thanks @sestevens ! I'll watch for the change.

We have another layer of abstraction in that we're using the SDK in a react-native mobile app so we end up leaning a little bit heavier on the Drop-in's presentation of methods.

Our current attempt at a temporary work around is hoping that the issue with the human interface standards was more one of "white button on light background" (which we've seen when adding Apple login method), so we're forcing in the dark theme instead of Dynamic in hopes that that passes the review.

**Comment by sestevens** [comment]
Sounds good!

**Comment by scannillo** [comment]
Hi @adamfellon - after some discussion within our team, we came to the conclusion that our docs need to better clarify how we expect developers to offer Apple Pay via Drop-In. 

> If your customer selected Apple Pay as their preferred payment method then result.paymentOptionType == .ApplePay and the result.paymentMethod will be nil. Selecting Apple Pay does not display the Apple Pay sheet or create a nonce - you will still need to do that at the appropriate time in your app.

_This comes from [the Apple Pay + Drop In section ](https://github.com/braintree/braintree-ios-drop-in#apple-pay--drop-in)in our README._

The Apple Pay offering via Drop-In is intended to function differently than the other payment flows (ex: PayPal). 

*  PayPal: The Drop-In handles the presentation of the PayPal checkout flow. It then returns back to the developer the `nonce` via the `BTDropInControllerHandler`.
*  Apple Pay: The Drop-In tells you when a customer _selects_ Apple Pay. You are responsible for inspecting the `result.paymentOptionType`, showing the Apple Pay button, and launching the Apple Pay sheet.

We think that Apple gave this rejection if you are launching the Apple Pay sheet directly from the callback block, without showing the Apple Pay button. Let us know if this sounds like your setup!

**Comment by adamfellon** [comment]
Good news, we got approved this morning after forcing the dark theme into the Drop-In.  🎉 

When Apple rejected us they sited the attached screenshot, and the human interface guidelines

![Screen Shot 2021-01-21 at 9 56 03 AM](https://user-images.githubusercontent.com/3448673/105376233-27431e00-5bcf-11eb-8879-e8c34c12ec53.png)

When we present the apple pay button with the other options available, but we were defaulting to the the dynamic colorScheme for ios versions > 13

```
        if (@available(iOS 13.0, *)) {
            BTUIKAppearance.sharedInstance.colorScheme = BTUIKColorSchemeDynamic;
        } else {
            BTUIKAppearance.sharedInstance.colorScheme = BTUIKColorSchemeDark;
        }
```

My thinking on the rejection is that they didn't like the white [ <logo>  Pay ] button on the light gray background, forcing it to white button on black background through just using `BTUIKColorSchemeDark` & not `BTUIKColorSchemeDynamic` was approved. 

Should that dynamic colorscheme also flip invert the applebutton colors to black when on a lighter background?  

Maybe there was a way I should have been managing color inversion in the presentation based on `systemBackgroundColor`?

Thanks again!

**Comment by scannillo** [comment]
👋  Hey @adamfellon. I'm glad your app got approved! However, I'm not sure the reason is the color scheme. 

The Apple Pay guidelines don't have detailed info on the appropriate background color for the mark, but if it's the same as the button guidelines it says a white logo with a black outline is OK on light background colors. Also, the white w/ black outline is the only mark that they provide.

It could be the case that some manual reviewers at Apple are more or less strict than others, and aren't consistently giving your app the same level of review. 

Can you let us know if you are or aren't showing the Apple Pay button before presenting the Apple Pay sheet?

**Comment by adamfellon** [comment]
> says a white logo with a black outline is OK on light background colors. Also, the white w/ black outline is the only mark that they provide.

Huh, weird I see that now..missed the white w/ border being ok, think we might've just gotten unlucky on the manual review then...after 21 days going back and forth glad we got through finally this morning 😓 

We don't show the apple pay button prior to the drop down, so maybe that's the direction we need to go moving forward.

**Comment by scannillo** [comment]
> We don't show the apple pay button prior to the drop down, so maybe that's the direction we need to go moving forward.

To confirm, you would need to show the Apple Pay button _after_ you [show the Drop-in](https://developers.braintreepayments.com/guides/drop-in/setup-and-integration/ios/v4#starting-drop-in).

**Comment by scannillo** [comment]
Hi everyone, we updated our [README to give more explicit instructions](https://github.com/braintree/braintree-ios-drop-in#apple-pay--drop-in) on how you should be presenting the Apple Pay button in your UI, in tandem with using our Drop-In UI. 

Also note - we just released [v9.0.0-beta1](https://github.com/braintree/braintree-ios-drop-in/releases/tag/9.0.0-beta1) of the Drop-in SDK. _Note:_ the linked Apple Pay instructions above will apply to both v8 and v9 integrations. For information on upgrading to v9, see our [Migration Guide](https://github.com/braintree/braintree-ios-drop-in/blob/master/V9_MIGRATION.md).
